### DISCUSSION TITLE
How to evaluate an expression tree with specific variable values?

### DISCUSSION THREAD

**Discussion by cannotcomplain** [pybamm-team/PyBaMM#2064] [Q&A] [answered]
Hello all, 

I'm fairly new to PyBaMM and am having an issue evaluating an expression tree with specific values substituted for variables. 

In this simple case, I am trying to extract the RHS of the ODE for SEI growth and get its value by processing it with parameters, but there remains a variable 'Outer SEI thickness' that I would like to substitute different values for (I understand I can just copy the expression but I'd like to automate for other examples as well. 

This is what I have done to get the expression tree I'd like to evaluate... 
```python
import pybamm
model = pybamm.lithium_ion.DFN({'SEI porosity change' : 'true', 'SEI' : 'solvent-diffusion limited'})
params = pybamm.ParameterValues('Chen2020_plating')
dLdt = params.process_symbol(list(model.submodels['sei'].rhs.values())[0])
```
which gives me the output
```python 
In[2]: dLdt
```
```
Out[2]: Multiplication(0x42dfa67e9f939aa8, *, children=['-827.1189588', '-1.733028823339507e-05 / Outer SEI thickness'], domains={'primary': ['negative electrode'], 'secondary': ['current collector']})
```

My question is how to substitute different values of 'Outer SEI thickness' into this expression? I have tried to update the `params` to include `'Outer SEI thickness'`as a variable but it still doesn't evaluate. 

Any help would be much appreciated. 

Thanks!

**Comment by valentinsulzer** [comment, marked as answer]
There is no straightforward way to do this, but a hacky way is 
```python
import pybamm
import numpy as np

model = pybamm.lithium_ion.DFN(
    {"SEI porosity change": "true", "SEI": "solvent-diffusion limited"}
)
params = pybamm.ParameterValues("Chen2020_plating")

# Discretize the model
sim = pybamm.Simulation(model, parameter_values=params)
sim.build()
# Extract dLdt
L = model.variables["Outer SEI thickness"]
L_disc = sim.built_model.variables["Outer SEI thickness"]
dLdt = sim.built_model.rhs[L]
dLdt.render()

# Build a y to evaluate
y = np.zeros(L_disc.y_slices[0].stop)
y[L_disc.y_slices[0]] = 1
print(dLdt.evaluate(y=y))
```
